### PR TITLE
DOC: Organize regressions

### PR DESCRIPTION
**Pull request by TomAugspurger**

### CODE REVIEW

File: doc/source/whatsnew/v1.0.2.rst [hunk 1]
```diff
@@ -15,23 +15,26 @@ including other versions of pandas.
...
 - Fixed regression in :meth:`rolling(..).corr() <pandas.core.window.rolling.Rolling.corr>` when using a time offset (:issue:`31789`)
 - Fixed regression in :meth:`groupby(..).nunique() <pandas.core.groupby.DataFrameGroupBy.nunique>` which was modifying the original values if ``NaN`` values were present (:issue:`31950`)
 - Fixed regression in ``DataFrame.groupby`` raising a ``ValueError`` from an internal operation (:issue:`31802`)
-- Fixed regression where :func:`read_pickle` raised a ``UnicodeDecodeError`` when reading a py27 pickle with :class:`MultiIndex` column (:issue:`31988`).
-- Fixed regression in :class:`DataFrame` arithmetic operations with mis-matched columns (:issue:`31623`)
 - Fixed regression in :meth:`groupby(..).agg() <pandas.core.groupby.GroupBy.agg>` calling a user-provided function an extra time on an empty input (:issue:`31760`)
-- Fixed regression in joining on :class:`DatetimeIndex` or :class:`TimedeltaIndex` to preserve ``freq`` in simple cases (:issue:`32166`)
-- Fixed regression in the repr of an object-dtype :class:`Index` with bools and missing values (:issue:`32146`)
+
```
Comment (jreback) [inline doc/source/whatsnew/v1.0.2.rst]: maybe organize how we do the bug fix categories? easier to read that way

Comment (TomAugspurger) [reply doc/source/whatsnew/v1.0.2.rst]: Done.

File: doc/source/whatsnew/v1.0.2.rst [hunk 2]
```diff
@@ -15,23 +15,34 @@ including other versions of pandas.
...
+**I/O**
+
 - Fixed regression in :meth:`read_csv` in which the ``encoding`` option was not recognized with certain file-like objects (:issue:`31819`)
-- Fixed regression in :meth:`DataFrame.reindex` and :meth:`Series.reindex` when reindexing with (tz-aware) index and ``method=nearest`` (:issue:`26683`)
+- Fixed regression in :meth:`DataFrame.to_excel` when the ``columns`` keyword argumnet is passed (:issue:`31677`)
 - Fixed regression in :class:`ExcelFile` where the stream passed into the function was closed by the destructor. (:issue:`31467`)
+- Fixed regression where :func:`read_pickle` raised a ``UnicodeDecodeError`` when reading a py27 pickle with :class:`MultiIndex` column (:issue:`31988`).
+
+**Reindexing**
```
Comment (jorisvandenbossche) [inline doc/source/whatsnew/v1.0.2.rst]:
```suggestion
**Reindexing/alignment**
```

**Review by jorisvandenbossche — APPROVED** [review]

File: doc/source/whatsnew/v1.0.2.rst [hunk 3]
```diff
@@ -15,23 +15,34 @@ including other versions of pandas.
...
-- Fixed regression in :class:`DataFrame` arithmetic operations with mis-matched columns (:issue:`31623`)
 - Fixed regression in :meth:`groupby(..).agg() <pandas.core.groupby.GroupBy.agg>` calling a user-provided function an extra time on an empty input (:issue:`31760`)
-- Fixed regression in joining on :class:`DatetimeIndex` or :class:`TimedeltaIndex` to preserve ``freq`` in simple cases (:issue:`32166`)
-- Fixed regression in the repr of an object-dtype :class:`Index` with bools and missing values (:issue:`32146`)
+
+**I/O**
+
 - Fixed regression in :meth:`read_csv` in which the ``encoding`` option was not recognized with certain file-like objects (:issue:`31819`)
-- Fixed regression in :meth:`DataFrame.reindex` and :meth:`Series.reindex` when reindexing with (tz-aware) index and ``method=nearest`` (:issue:`26683`)
+- Fixed regression in :meth:`DataFrame.to_excel` when the ``columns`` keyword argumnet is passed (:issue:`31677`)
```
Comment (simonjayhawkins) [inline doc/source/whatsnew/v1.0.2.rst]:
```suggestion
- Fixed regression in :meth:`DataFrame.to_excel` when the ``columns`` keyword argument is passed (:issue:`31677`)
```

**Comment by TomAugspurger** [comment]
I'm not going to worry about the build that failed to start as long as the doc build finishes.

**Pull request merged**